pkgs/applications/science/machine-learning/vowpal-wabbit/default.nix

```diff
@@ -1,42 +1,42 @@
-{ lib, stdenv, fetchFromGitHub, cmake, boost, flatbuffers, rapidjson, spdlog, zlib }:
+{ lib, stdenv, fetchFromGitHub, fetchpatch, cmake, boost, eigen, rapidjson, spdlog, zlib }:
 
 stdenv.mkDerivation rec {
   pname = "vowpal-wabbit";
-  version = "9.0.1";
+  version = "9.6.0";
 
   src = fetchFromGitHub {
     owner = "VowpalWabbit";
     repo = "vowpal_wabbit";
     rev = version;
-    sha256 = "sha256-ZUurY2bmTKKIW4GR4oiIpLxb6DSRUNJI/EyNSOu9D9c=";
+    sha256 = "sha256-iSsxpeTRZjIhZaYBeoKLHl9j1aBIXWjONmAInmKvU/I=";
   };
 
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/VowpalWabbit/vowpal_wabbit/commit/0cb410dfc885ca1ecafd1f8a962b481574fb3b82.patch";
+      sha256 = "798246d976932a6e278d071fdff0bff57ba994a3aef58926bbf4e0bf6fa09690";
```

Comment:
How did you get this hash? Ofborg says there is hash mismatch: https://github.com/NixOS/nixpkgs/pull/200407/checks?check_run_id=9735436805

Reply:
I reused the hash that was calculated when we applied the [patch in homebrew](https://github.com/Homebrew/homebrew-core/blob/c79b4baa83c7c0670c31e5f8245a3d71d5ec548c/Formula/vowpal-wabbit.rb#L27). However, I removed the `?full_index=1` url parameter and didn't realize the hash changed which of course makes sense since the file contents changed. I'll update the hash of the patch.

Reply:
TBH I know nothing about fetchpatch's internals, but I guess `?full_index=1` won't change the sha256 in this case -- fetchpatch should likely strip that part before calculating the hash. I will simply suggest the "trust on first use" way to get the correct hash -- give a wrong hash to fetchpatch, try to build and get the correct hash from the error log.

Reply:
It looks like `?full_index=1` causes the patch itself to have full commit refs instead of shortened ones. So it makes sense the hash of the contents changes. The hash looks correct now and the package should be good to go now hopefully.